cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -256,14 +256,16 @@ struct GroupByNode : ExecNode {
               const std::vector<int>&& key_field_ids,
               const std::vector<int>&& agg_src_field_ids,
               const std::vector<internal::Aggregate>&& aggs,
-              const std::vector<const HashAggregateKernel*>&& agg_kernels)
+              const std::vector<const HashAggregateKernel*>&& agg_kernels,
```

Comment:
If you wouldn't mind cleaning this up:
```suggestion
              std::vector<const HashAggregateKernel*> agg_kernels,
```
also ExecNode subclasses shouldn't be `struct`s